### PR TITLE
Added api endpoints for course, submission and event

### PR DESCRIPTION
**Pull request by seveibar**
This should get us started for course, submission and api endpoints, these are necessary to start virtually any backend issue

### CODE REVIEW

**Comment by seveibar** [comment]
This is mostly generator except for the model files, please check all of those out- that's the important stuff

**Comment by seveibar** [comment]
These were the generator commands used, the resulting files (atleast the model.js) was modified so make sure to review the three model.js files...

```
 yo angular-fullstack:endpoint event
 yo angular-fullstack:endpoint course
 yo angular-fullstack:endpoint submission
```

File: server/api/course/course.model.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+var mongoose = require('bluebird').promisifyAll(require('mongoose'));
+var Schema = mongoose.Schema;
+
+var CourseSchema = new Schema({
+  name: String,
+  description: String,
+  students: [{type : Schema.Types.ObjectId, ref: 'User'}],
+  active: Boolean
```
Comment (kmcnellis) [inline server/api/course/course.model.js:15]: Should probably associate an instructor with the course

File: server/api/event/event.model.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  title: String,
+  description: String,
+  course: {type: Schema.Types.ObjectId, ref: 'Course'},
+  submissions: [{type : Schema.Types.ObjectId, ref: 'Submission '}],
+  locations: [String], // TODO, what format should this be in?
+  times: [{
+      start: Date,
+      end: Date
+  }]
+});
```
Comment (kmcnellis) [inline server/api/event/event.model.js:18]: should record info about who created it and when

File: server/api/course/course.model.js (hunk 1 above)
Comment (seveibar) [inline server/api/course/course.model.js:15]: good call

File: server/api/submission/submission.model.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+var mongoose = require('bluebird').promisifyAll(require('mongoose'));
+var Schema = mongoose.Schema;
+
+var SubmissionSchema = new Schema({
+  title: String,
+  content: String,
+  images: [String], // path to image on static image server?
+  time: Date,
+  authors: [{type : Schema.Types.ObjectId, ref: 'User'}],
```
Comment (kmcnellis) [inline server/api/submission/submission.model.js:12]: Should distinguish user who submitted it from people who are claimed to be in the image (those checking in)

File: server/api/event/event.model.js (hunk 2 above)
Comment (seveibar) [inline server/api/event/event.model.js:18]: is this redundant with course? course has an instructor- does it matter which instructor created the event- or are there student-generated events?

File: server/api/submission/submission.model.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  location: String // TODO format?
```
Comment (kmcnellis) [inline server/api/submission/submission.model.js]: I believe there is a mongodb location format

File: server/api/submission/submission.model.js (hunk 3 above)
Comment (seveibar) [inline server/api/submission/submission.model.js:12]: `submitter` good call

File: server/api/submission/submission.model.js (hunk 4 above)
Comment (seveibar) [inline server/api/submission/submission.model.js]: like a gps coordinate? we might also want a location description... I'll add that as well

File: server/api/event/event.model.js (hunk 2 above)
Comment (kmcnellis) [inline server/api/event/event.model.js:18]: If an admin creates an event or there are multiple instructors or the instructors switch.  Just for general papertrails and record-keeping.

File: server/api/event/event.model.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+var mongoose = require('bluebird').promisifyAll(require('mongoose'));
+var Schema = mongoose.Schema;
+
+var EventSchema = new Schema({
+  title: String,
+  description: String,
+  course: {type: Schema.Types.ObjectId, ref: 'Course'},
```
Comment (kmcnellis) [inline server/api/event/event.model.js]: Shouldn't an event be able to be linked to multiple courses?

Comment (seveibar) [inline server/api/event/event.model.js]: good call- `courses`

File: server/api/event/event.model.js (hunk 2 above)
Comment (seveibar) [inline server/api/event/event.model.js:18]: alright makes sense

File: server/api/submission/submission.model.js (hunk 4 above)
Comment (kmcnellis) [inline server/api/submission/submission.model.js]: Probably an address, coordinates, and description

Comment (seveibar) [inline server/api/submission/submission.model.js]: something like?

```
location: {
    address: String,
    coordinates: ???,
    description: String
}
```

Comment (kmcnellis) [inline server/api/submission/submission.model.js]: Sure.  The mongo location type has a lot of cool queries we can use. https://docs.mongodb.org/manual/applications/geospatial-indexes/

File: server/api/course/course.model.js (hunk 1 above)
Comment (jowlee) [inline server/api/course/course.model.js:15]: Add field for CRN(Course reference number) and Dept. so it's easier to search

Comment (seveibar) [inline server/api/course/course.model.js:15]: also good call

Comment (kmcnellis) [inline server/api/course/course.model.js:15]: Also course number and semester

Comment (jowlee) [inline server/api/course/course.model.js:15]: Should there be a field for TA's ?

Comment (seveibar) [inline server/api/course/course.model.js:15]: @leej42 I think that's a good idea- but wasn't in the original spec- so i'm going to recommend we go sort of MVP style

**Pull request merged**